UnityProject/Assets/Scripts/Objects/Doors/DoorMasterController.cs

```diff
@@ -112,6 +122,28 @@ private void Awake()
 			apc = GetComponent<APCPoweredDevice>();
 			doorAnimator = GetComponent<DoorAnimatorV2>();
 			doorAnimator.AnimationFinished += OnAnimationFinished;
+			LoadManager.RegisterActionDelayed(DelayedRegister, 2);
+		}
+
+		public void DelayedRegister()
+		{
+			HackingProcessBase.RegisterPort(TryForceClose, this.GetType());
+			HackingProcessBase.RegisterPort(tryBump, this.GetType());
+			HackingProcessBase.RegisterPort(TryClose, this.GetType());
+			HackingProcessBase.RegisterPort(CheckPower, this.GetType());
+			HackingProcessBase.RegisterPort(ConfirmAIConnection, this.GetType());
+		}
+
+		public bool GetPowerState()
+		{
+			PowerState = false;
+			HackingProcessBase.ImpulsePort(CheckPower);
```

Comment:
I don't understand this one, why we need two methods to get the power state?

Reply:
it's that wraps the PowerState = false; and ImpulsePort into one function, ImpulsePort isn't supposed to be called directly